src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/components/TradeQuoteContent.tsx

```diff
@@ -100,6 +100,23 @@ export const TradeQuoteContent = ({
     }
   }, [isHighPriceImpact, isModeratePriceImpact, priceImpactPercentage, translate])
 
+  const eta = useMemo(() => {
+    if (totalEstimatedExecutionTimeMs === undefined) return null
+
+    return (
+      <Skeleton isLoaded={!isLoading}>
+        <Flex gap={2} alignItems='center'>
+          <RawText color='text.subtle'>
+            <FaRegClock />
+          </RawText>
+          {totalEstimatedExecutionTimeMs === 0
+            ? '0s'
+            : prettyMilliseconds(totalEstimatedExecutionTimeMs)}
+        </Flex>
+      </Skeleton>
+    )
+  }, [totalEstimatedExecutionTimeMs, isLoading])
+
```

Comment:
Extracted into its own constant for sanity:

- undefined = no dice, we no know 
- 0 = 0s (this doesn't get parsed nicely by `prettyMilliseconds` and will just render 0 instead of 0s)
- >0 = lfg